### PR TITLE
Add native timer clean-up behavior

### PR DESCRIPTION
**Pull request by JadenSimon**
#### Purpose (TL;DR) - mandatory
See #406 for context.

Adds `shouldHandleNativeTimers` flag to the config (set to false by default). When enabled, calls to `clear/cancel` are forwarded to their real functions if the timer was not created by `FakeTimers`.

### CODE REVIEW

File: src/fake-timers-src.js [hunk 1]
```diff
@@ -746,6 +750,21 @@ function withGlobal(_global) {
         // its .id field is the actual timer id.
         const id = typeof timerId === "object" ? timerId.id : timerId;
 
+        // If `id` is undefined then we can be pretty sure the timer is not from us
```
[the remark this answers is not in the document]
Comment (JadenSimon) [reply src/fake-timers-src.js:753]: Hey thanks for the update! I've mostly just been busy with other stuff. Was planning on making a commit w/ feedback but got sidetracked. Should be able to update this PR sometime today with the suggested fix for the Node timers.

Adding the symbol should work fine, don't see any reason why it wouldn't work.

File: src/fake-timers-src.js [hunk 2]
```diff
@@ -746,6 +750,21 @@ function withGlobal(_global) {
...
+        if (id === undefined || id < idCounterStart) {
+            if (clock.shouldHandleNativeTimers === true) {
+                const nativeHandler = clock[`_${handlerName}`];
+                return typeof nativeHandler === "function"
+                    ? nativeHandler(timerId)
+                    : undefined;
+            }
+            // eslint-disable-next-line no-console
+            console.warn(
```
[the remark this answers is not in the document]
Comment (benjamingr) [reply src/fake-timers-src.js:762]: Sorry this was missed - yes this sounds fine - we don't have a "standard" way to emit warnings so I guess `process.emitWarning` in Node.js and a `console.warn` in browsers?

File: src/fake-timers-src.js [hunk 3]
```diff
@@ -730,8 +733,9 @@ function withGlobal(_global) {
      * @param {Clock} clock
      * @param {number} timerId
      * @param {string} ttype
+     * @param {string} handlerName
      */
-    function clearTimer(clock, timerId, ttype) {
+    function clearTimer(clock, timerId, ttype, handlerName) {
```
[the remark this answers is not in the document]
Comment (JadenSimon) [reply src/fake-timers-src.js]: Didn't seem to work. I just gave the timer the corresponding type. So now all timers have a reference to how they were created.

**Review by benjamingr — APPROVED** [review]
Looks good to me, @fatso83 wdyt?

File: src/fake-timers-src.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (benjamingr) [reply src/fake-timers-src.js]: Yeah the toPrimitive changes are good and we missed those - Node added support for that a while back

[the remark this answers is not in the document]
Comment (fatso83) [reply src/fake-timers-src.js]: I checked before writing: it works from Node 12 and onwards (not Node 0.10.x).

**Review by fatso83 — APPROVED** [review]
Looks very fine to my eyes 😍

**Pull request merged**